src/handle_provide_parameter.c

```diff
@@ -727,11 +731,49 @@ static void handle_execute(ethPluginProvideParameter_t *msg, context_t *context)
             }
             break;
 
+            // ###################
+            // Parsing PAY_PORTION
+            // ###################
+
+        case INPUT_PAY_PORTION_LENGTH:
+            context->next_param = INPUT_PAY_PORTION_TOKEN;
+            break;
+        case INPUT_PAY_PORTION_TOKEN:
+            context->next_param = INPUT_PAY_PORTION_RECIPIENT;
+            break;
+        case INPUT_PAY_PORTION_RECIPIENT:
+            context->next_param = INPUT_PAY_PORTION_AMOUNT;
+            break;
+        case INPUT_PAY_PORTION_AMOUNT:
+            // empty statement because switch / case is wonky
+            ;
+            uint8_t decoding[sizeof(context->pay_portion_amount)];
+            for (uint8_t i = 0; i < sizeof(decoding); ++i) {
+                decoding[i] = msg->parameter[PARAMETER_LENGTH - 1 - i];
+            }
+            uint16_t decoded;
+            memmove(&decoded, decoding, sizeof(context->pay_portion_amount));
+            uint32_t pay_portion_sum = context->pay_portion_amount + decoded;
```

Comment:
I don't knowthe plugin itself, may be my comment is naive, but here, there is an addition, whrere one of the components comes within an array??

Reply:
Both are uint16